docs/adr/0029-server-side-caching.md

```diff
@@ -0,0 +1,46 @@
+# Server Side Caching
```

Comment:
Just a point of style - most of the ADRs seem to put the decision in the title, not just the subject. So for this, probably something like "Use in-memory server-side caching".

Reply:
Starting review